mars/oscar/backends/message.py

```diff
@@ -191,6 +204,17 @@ def __init__(
     def message_type(self) -> MessageType:
         return MessageType.error
 
+    def as_instanceof_cause(self):
+        if issubclass(self.error_type, ErrorMessage.AsCauseBase):
+            return self.error.with_traceback(self.traceback)
+
+        class _MarsError(ErrorMessage.AsCauseBase, type(self.error)):
+            pass
+
+        return _MarsError(
```

Comment:
Wrapping existing errors has some side effects. For instance, some exception types have custom attributes, for instance, HTTP error codes. Wrapping with error messages hide these attributes and may lead to unexpected errors. Simply try adding attributes to these exceptions and log locations when handling exceptions at receiver end may be a better solution.

Reply:
Fixed. Thanks. I added a test case `test_as_instanceof_cause` for testing `as_instanceof_cause`.